haskellPackages: update stackage and hackage

maralorn opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/Mw5WLnzsP7fC4Zky) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #157583. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

maralorn committed:
haskellPackages: stackage-lts 18.23 -> 18.24

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

maralorn committed:
all-cabal-hashes: 2022-01-31T19:23:21Z -> 2022-02-06T15:34:38Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

maralorn committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages: preserve ghc-exactprint 1.4.1

We'll continue using this version for GHC 9.2.1. 1.5.0 changes behavior
in a breaking way, but the API stays the same, making it a risky
upgrade. Hopefully dependents will tell us via version constraints what
to use in the future.

github-actions[bot] committed:
Merge master into haskell-updates

dalpd committed:
haskellPackages.numerals: Disable version checks

We could also be using fetchFromGithub but the repo looks stale enough that it
doesn't really matter, plus the upper bounds for almost all the dependencies
doesn't exist anymore.

sternenseemann committed:
haskell.packages.ghc902.alex: drop unnecessary patch

sternenseemann commented:
HLS started to link dynamically against other libraries and GHC itself: https://github.com/haskell/haskell-language-server/commit/3f12824f0f1776f7935a1652e09f02a019c86f65

I'm not sure if this means anything for closure size, in theory it looks like a welcome change in terms of functionality, but maybe doesn't solve a huge problem for us, as GHC, the libraries and system dependencies would usually match in a nix development environment (although I am pretty sure that there are a lot of non-usual cases as well).

In any case we need to prevent GHC from adding `/build/*/dist/build` to RPATH for GHC 9.2.1. Currently trying to figure out if this is a new regression in 9.2.1 or something that has always been a problem with GHC.

sternenseemann committed:
haskellPackages.haskell-language-server: fix shared build

sternenseemann committed:
haskellPackages.ihaskell: use enableSharedExecutable over adhoc flag

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

sternenseemann committed:
haskellPackages.hs-mesos: place dont-distribute entry appropriately

hs-mesos doesn't depend on webkit and was misplaced.

sternenseemann committed:
haskell-language-server: make linking configureable in wrapper

haskell-language-server will now default to building a shared
executable, as upstream does, complete with a huge closure. By passing
{ dynamic = false; } via override, it is still possible to build a
"statically linked" variant of HLS, as it used to be.

Note: Before this change HLS would fail to compile on aarch64.

sternenseemann committed:
haskell-language-server: retain old postInstall if any

sternenseemann committed:
haskell-language-server: default to statically linking haskell deps

This means we don't use the system linker by default, but decreases
closure size significantly, as we no longer reference every supported
version GHC.

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell-language-server: fix reference logic

If we are linking dynamically, it's practically no use removing
references, as we depend on GHC either way via linking.

I've also elected to keep the references to the data outputs in all
cases — they are a bit arcane (there's no easy way to tell they
definitely are not necessary) and don't contribute too much to the
overall closure size.

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
Merge pull request #158669 from sternenseemann/hls-dynamic-wrapper

haskell-language-server: make linking configureable in wrapper

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh

sternenseemann commented:
HLS failure on aarch64-linux was flaky, as I've been able to confirm.

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates
